library/src/com/google/maps/android/kml/KmlContainer.java

```diff
@@ -0,0 +1,254 @@
+        return mStyles;
+    }
+
+    /**
+     * Takes an ArrayList of styles and assigns these folders with the styles
+     */
+    /* package */ void setStyles(HashMap<String, KmlStyle> styles) {
+        mStyles = styles;
+    }
+
+    /**
+     * @param placemarks Placemark for the container to contain
+     * @param object     Corresponding GoogleMap map object of the basic_placemark (if it has been
+     *                   added
+     *                   to the map)
+     */
+    /* package */ void setPlacemark(KmlPlacemark placemarks, Object object) {
+        mPlacemarks.put(placemarks, object);
+    }
+
+    /**
+     * Add a nested container
+     *
+     * @param container Container to nest within the current instance of the container
+     */
+    /* package */ void addChildContainer(KmlContainer container) {
+        mContainers.add(container);
+    }
+
+    /**
+     * @return HashMap of containers
+     */
+    public HashMap<KmlPlacemark, Object> getPlacemarks() {
+        return mPlacemarks;
+    }
+
+    /**
+     * Sets a property to be contained by the container
+     *
+     * @param propertyName  Name of the property, ie "name"
+     * @param propertyValue Value of the property, ie "Arizona"
+     */
+    public void setProperty(String propertyName, String propertyValue) {
+        mProperties.put(propertyName, propertyValue);
+    }
+
+    /**
+     * Sets a style to be contained by the container
+     *
+     * @param styleId Name or ID of the style
+     * @param style   KmlStyle object
+     */
+    public void setStyle(String styleId, KmlStyle style) {
+        mStyles.put(styleId, style);
+    }
+
+    /**
+     * @return A map of strings representing a style map, null if no style maps exist
+     */
+    /* package */ HashMap<String, String> getStyleMap() {
+        return mStyleMap;
+    }
+
+    /**
+     * @param styleMap Adds a map of strings representing a style map
+     */
+    /* package */ void setStyleMap(HashMap<String, String> styleMap) {
+        mStyleMap.putAll(styleMap);
+    }
+
+    /**
+     * Add a ground overlay for this container
+     *
+     * @param groundOverlay ground overlay to add
+     */
+    /* package */ void addGroundOverlay(KmlGroundOverlay groundOverlay) {
+        mGroundOverlays.put(groundOverlay, null);
+    }
+
+    /**
+     * Gets all of the ground overlays which were set in the container
+     *
+     * @return  A set of ground overlays
+     */
+    /* package */ HashMap<KmlGroundOverlay, GroundOverlay> getGroundOverlayHashMap() {
+        return mGroundOverlays;
+    }
+
+    /**
+     * Gets the value of a property based on the given key
+     *
+     * @param propertyName property key to find
+     * @return value of property found, null if key doesn't exist
+     */
+    public String getKmlProperty(String propertyName) {
+        return mProperties.get(propertyName);
+    }
+
+    /**
+     * Gets whether the container has any properties
```

Comment:
This reads a little strangely. How about "Determines if the container has properties."